app/controllers/encrypt_controller.rb

```diff
@@ -179,6 +178,44 @@ def stats
     render json: success_json.merge(encrypted_pms_count: [pms_count, keys_count].max)
   end
 
+  # Get all topic keys for current user.
+  #
+  # Returns all keys to all encrypted PMs.
+  def show_all_keys
+    encrypted_topic_users = EncryptedTopicsUser.where(user: current_user)
+    topic_keys = encrypted_topic_users.map { |etu| [etu.topic_id, etu.key] }.to_h
```

Comment:
This could be done more efficiently with `pluck`

```suggestion
    topic_keys = EncryptedTopicsUser.where(user: current_user).pluck(:topic_id, :key).to_h
```